### PR TITLE
Save

### PR DESCRIPTION
**Pull request by KevinRoig**

### CODE REVIEW

**Review by AurelieFqt — DISMISSED** [review]

File: src/Controller/GalleryController.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class GalleryController extends AbstractController
+{
+    /**
+     * @Route("/gallerie-design-metallique/", name="gallery_metallicDesign")
+     */
+    public function metallicDesignGallery(): Response
+    {
+        $images = $this->getDoctrine()
+        ->getRepository(Image::class)
+        ->findAll();
```
Comment (RomainClair) [inline src/Controller/GalleryController.php]: Use Dependency injection to get Services, it's the symfony best practice : https://symfony.com/doc/current/best_practices.html#use-dependency-injection-to-get-services

File: src/Controller/MainFormController.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    /**
+     * @Route("/form/", name="form")
+     */
+    public function index(Request $request): Response
+    {
+        $image = new Image();
+        $form = $this->createForm(MainFormType::class, $image);
+        $form->handleRequest($request);
+        if ($form->isSubmitted()) {
+            $entityManager = $this->getDoctrine()->getManager();
```
Comment (RomainClair) [inline src/Controller/MainFormController.php]: Same here, the entityManager should be a parameter

File: src/Entity/Article.php [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    private string $section;
+
+    /**
+     * @ORM\ManyToMany(targetEntity=Image::class, inversedBy="articles")
+     */
+    private collection $image;
```
Comment (RomainClair) [inline src/Entity/Article.php]: It probably should be Collection (with capital C because it's a class)

File: src/Form/MainFormType.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+
+class MainFormType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('nom', TextType::class)
+            ->add('url', TextType::class)
```
Comment (RomainClair) [inline src/Form/MainFormType.php:17]: Don't you want to add images using upload instead of asking URLs ?

**Review by RomainClair — CHANGES_REQUESTED** [review]
Identifiers should probably be in english to help collaboration with any dev

File: src/Entity/Article.php (hunk 3 above)
Comment (KevinRoig) [reply src/Entity/Article.php]: Ok, check.

File: src/Controller/MainFormController.php (hunk 2 above)
Comment (KevinRoig) [reply src/Controller/MainFormController.php]: I am not sure about that but in the documentation,it is exactly like i did! https://symfony.com/doc/current/forms.html#processing-forms

File: src/Controller/GalleryController.php (hunk 1 above)
Comment (KevinRoig) [reply src/Controller/GalleryController.php]: OK, changed.

File: src/Controller/MainFormController.php (hunk 2 above)
Comment (RomainClair) [reply src/Controller/MainFormController.php]: It is working and i do believe lots of developers will do it that way, specially when they use Symfony for a _long_ time.
But the best practices Symfony recommend is to use dependecy injection for getting services.
Like the EntityManager.
Many examples in the Symfony documentation still uses getDoctrine()->getManager
But this one https://symfony.com/doc/current/doctrine.html#persisting-objects-to-the-database show how to do it (but in comments)

Comment (KevinRoig) [reply src/Controller/MainFormController.php]: Ok, i see it know! 
Modification done on next push.

File: src/Form/MainFormType.php (hunk 4 above)
Comment (KevinRoig) [reply src/Form/MainFormType.php:17]: Ok, we'll have to work on it (tryed this morning but didn't work).
Can you validate the rest of the PR so everyone gets it for the holidays? (name fields will be change after we make the form work)

**Review by AurelieFqt — APPROVED** [review]

**Review by RomainClair — APPROVED** [review]

**Pull request merged**